src/libclipper/src/json_util.cpp

```diff
@@ -384,6 +384,33 @@ std::string to_json_string(rapidjson::Document& d) {
   return buffer.GetString();
 }
 
+void set_string_array(rapidjson::Document& d,
+                      const std::vector<std::string>& str_array) {
+  d.SetArray();
+  for (auto str : str_array) {
```

Comment:
This should be `for (auto const &str: str_array) {`. The reason being that using `auto` alone iterates by value, which I'm pretty sure incurs a copy.